slither/core/declarations/structure.py

```diff
@@ -51,3 +51,17 @@ def elems_ordered(self) -> List["StructureVariable"]:
 
     def __str__(self) -> str:
         return self.name
+
+    def __eq__(self, other) -> bool:
+        if not isinstance(other, Structure):
+            return False
+        if len(self.elems) != len(other.elems):
+            return False
+        for idx, elem in enumerate(self.elems_ordered):
+            other_elem = other.elems_ordered[idx]
+            if str(other_elem.type) != str(elem.type) or other_elem.name != elem.name:
```

Comment:
I think this might create issue where two types share the same name

For example in 
```solidity
struct St{
	uint a;
}

contract A{

	St s;

}
```

```solidity
import {A} from "test.sol";

struct St{
	uint b;
}

contract B is A{
	St s2;
}
```

Maybe we should create a `canonical_name` for type that would include the source mapping definition?  (cc @smonicas  @0xalpharush)

Reply:
Seems like `canonical_name` would be a good idea. But in the example you gave, wouldn't it be correct to say that the two `St` structs are not equal? In the highlighted code, while enumerating the elements of the two structs, the `other_elem.name != elem.name` would cause my `__eq__()` implementation to return `False`, which seems correct at a glance. Granted, comparing canonical names would probably be better still.

Reply:
I suppose if you renamed `St.b` in the second code block to `uint a`, then the resulting equality might be considered a false positive, since it would see that both structs have the same name, and their elements have the same types and names, therefore returning true. But that seems quite picky, since the two structs would be identical except for where they are defined in Solidity.